### PR TITLE
Create wine price parsers 

### PR DESCRIPTION
**Pull request by antos07**

### CODE REVIEW

File: backend/wine_map/serializers.py [hunk 1]
```diff
@@ -38,3 +38,10 @@ class CommentSerializer(serializers.ModelSerializer):
     class Meta:
         model = Comment
         fields = "__all__"
+
+
+class WineInShopSerializer(serializers.Serializer):
+    shop_name = serializers.CharField()
+    min_price = serializers.FloatField()
+    max_price = serializers.FloatField()
```
Comment (Nattalli) [inline backend/wine_map/serializers.py:46]: in one shop can be two prices?

File: backend/wine_map/views.py [hunk 2]
```diff
@@ -8,10 +9,12 @@
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
+from . import parsers
 from .filters import WineFilter
 from .models import Country, Brand, Wine, Comment
 from .permissions import IsCommentAuthor
-from .serializers import CategoriesSerializer, WineSerializer, CommentSerializer
+from .serializers import CategoriesSerializer, WineSerializer, CommentSerializer, \
+    WineInShopSerializer
```
Comment (Nattalli) [inline backend/wine_map/views.py:17]: better use `()` instead of `\`

File: backend/wine_map/views.py [hunk 3]
```diff
@@ -86,3 +89,14 @@ class CommentUpdateView(generics.UpdateAPIView):
...
+
+class WineInShopsView(APIView):
+    permission_classes = [AllowAny]
+
+    @extend_schema(responses=WineInShopSerializer(many=True))
+    def get(self, request: Request, wine_id: int) -> Response:
+        wine = get_object_or_404(Wine, pk=wine_id)
+        parsed_shops = parsers.parse_all(wine.name)
+        serializer = WineInShopSerializer(parsed_shops, many=True)
+        return Response(serializer.data)
```
Comment (Nattalli) [inline backend/wine_map/views.py:102]: add message, please

Comment (Nattalli) [reply backend/wine_map/views.py:102]: and status

File: backend/wine_map/parsers/silpo.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import json
+import typing
+from typing import Optional
+from urllib.parse import urlencode, quote
+
+import requests
+
+from . import errors
```
Comment (Nattalli) [inline backend/wine_map/parsers/silpo.py:8]: how ir works?

File: backend/wine_map/parsers/silpo.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                                 timeout=REQUEST_TIMEOUT)
+        response.raise_for_status()
+        data = response.json()
+    except (requests.RequestException, json.JSONDecodeError) as e:
+        raise errors.NetworkError(f"Could not fetch data for wine {wine_name}") from e
+    return data
+
+
+def parse_data(wine_name: str, data: dict[str, typing.Any]) -> Optional[WineInShop]:
+    items = data.get("items")
```
Comment (Nattalli) [inline backend/wine_map/parsers/silpo.py:36]: `items` mean wines? If it is - better use `wines` variable name

File: backend/wine_map/parsers/utils.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        silpo_future = executor.submit(parse_silpo, wine_name)
+        vinoua_future = executor.submit(parse_vinoua, wine_name)
+        winetime_future = executor.submit(parse_winetime, wine_name)
+
+        results = [_get_future_result_or_none(silpo_future),
+                   _get_future_result_or_none(vinoua_future),
+                   _get_future_result_or_none(winetime_future)]
+        results = [result for result in results if result]
+        return results
```
Comment (Nattalli) [inline backend/wine_map/parsers/utils.py:20]:
```suggestion
        return [result for result in results if result]
```

Comment (Nattalli) [reply backend/wine_map/parsers/utils.py:20]: just because you use `results` variable name twice

**Review by Nattalli — CHANGES_REQUESTED** [review]
Overall - very good job

File: backend/wine_map/serializers.py (hunk 1 above)
Comment (antos07) [reply backend/wine_map/serializers.py:46]: As I am parsing search results, I can't know which wine is the one I need

File: backend/wine_map/parsers/silpo.py (hunk 4 above)
Comment (antos07) [reply backend/wine_map/parsers/silpo.py:8]: Just a relative import. It imports the module `errors` from the package `wine_map.parsers`

**Review by Nattalli — APPROVED** [review]

**Review by Afformativ — APPROVED** [review]

**Pull request merged**